src/amr/data/particles/refine/split_2d.h

```diff
@@ -0,0 +1,481 @@
+                      Color::PURPLE>
+{
+};
+
+
+/**************************************************************************/
+template<>
+class Splitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<16>>
+    : public ASplitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<16>>
+{
+    using Dispatcher              = ExactSplitParticleDispatcher<DimConst<2>>;
+    static constexpr uint8_t root = 4;
+
+public:
+    Splitter() { split(weights_, deltas_); }
+
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, Deltas& deltas)
+    {
+        split(weights, weight_vals_, deltas, delta_vals_);
+    }
+
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, float const weight_vals[], //
+                      Deltas& deltas, float const delta_vals[])
+    {
+        for (size_t i : {0, 3, 12, 15}) // first loop = top left corner
+            weights[i] = weight_vals[ColorIndex::RED];
+
+        for (size_t i : {1, 2, 4, 7, 8, 11, 13, 14})
+            weights[i] = weight_vals[ColorIndex::ORANGE];
+
+        for (size_t i : {5, 6, 9, 10})
+            weights[i] = weight_vals[ColorIndex::GREEN];
+
+        Dispatcher::set_deltas(deltas, delta_vals, root);
+    }
+
+protected:
+    enum ColorIndex : uint8_t { GREEN = 0, ORANGE, RED };
+    constexpr static float delta_vals_[]  = {-1.5, -.5, .5, 1.5};
+    constexpr static float weight_vals_[] = {
+        0.140625, // 0 == GREEN
+        0.046875, // 1 == ORANGE
+        0.015625  // 2 == RED
+    };
+};
+
+
+/**************************************************************************/
+template<>
+class Splitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<4>, Color::PURPLE>
+    : public ASplitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<4>>
+{
+    using Dispatcher = SplitParticleDispatcher<DimConst<2>, RefinedParticlesConst<4>>;
+
+public:
+    Splitter() { split(weights_, deltas_); }
+
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, Deltas& deltas)
+    {
+        split(weights, weight_val_, deltas, delta_val_);
+    }
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, float const& weight_val, //
+                      Deltas& deltas, float const& delta_val)
+    {
+        Dispatcher::set_weights(weights, weight_val);
+        Dispatcher::set_deltas(deltas, delta_val);
+    }
+
+protected:
+    constexpr static float delta_val_  = 0.683734;
+    constexpr static float weight_val_ = 0.25;
+};
+
+
+template</*default pattern*/>
+class Splitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<4>>
+    : public Splitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<4>, Color::PURPLE>
+{
+};
+
+
+/**************************************************************************/
+template<>
+class Splitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<8>, Color::PINK, Color::PURPLE>
```

Comment:
I agree with these colors, but where are they used in the class ?

Reply:
they're not, they're to differentiate the structure from others